File: 2024-04-invert-text-case/invert_case.py

def invert_case(text: str):
    opposites = {}
    for (u, l) in zip(iter(text.upper()), iter(text.lower())):
        opposites[u] = l
        opposites[l] = u
    return "".join(opposites[c] for c in iter(text))

File: 2024-04-invert-text-case/invert_case.rs
/// Convert upper case to lower case, and vice-versa.
fn invert_case(text: &str) -> String {
    let mut opposites = std::collections::HashMap::<char, char>::new();
    for (l, u) in (text.to_lowercase().chars()).zip(text.to_uppercase().chars()) {
        opposites.insert(u, l);
        opposites.insert(l, u);
    }

    text.chars().map(|c| opposites[&c]).collect()
}
